### PR TITLE
added getCenterOfActiveScreen in Screen class

### PR DESCRIPTION
**Pull request by danielpetrica**
Added a function that Returns the center of the screen where the mouse pointer is placed. Useful if you want to initialize a smaller window and place it exactly in the center of screen.

Closes this discussion idea #373 

I ran pest and all tests passed. Also ran Phpstan on the specific file. Maybe it could be a good idea to add a test for it but don't know how.

### CODE REVIEW

**Comment by simonhamp** [comment]
I'd like to see part of this exposed via our Electron web service instead of being fully implemented here as Electron already supports some of what you're doing here. For example, you can use:
- [`getCursorScreenPoint`](https://www.electronjs.org/docs/latest/api/screen#screengetcursorscreenpoint) to get the cursor position ([this is what we use](https://github.com/NativePHP/electron-plugin/blob/main/src/server/api/screen.ts#L18) for the `cursorPosition` method) and then pass it to
- [`getDisplayNearestPoint`](https://www.electronjs.org/docs/latest/api/screen#screengetdisplaynearestpointpoint) to get the display that intersects with that `Point`.

I'm imagining a `Screen::active()` method which returns the details of the currently active [display](https://www.electronjs.org/docs/latest/api/structures/display).

This would be a useful primitive for other use-cases, not only for finding the center of the "current" screen.

Are you up for adapting this PR and creating the Electron side @danielpetrica?

**Comment by danielpetrica** [comment]
@simonhamp Sounds like a better and cleaner idea. At the time I didn't have much confidence on working on electron. I'll try to look into it as this would simplify my work over time too

**Comment by simonhamp** [comment]
No worries. [I've created the Electron side for you](https://github.com/NativePHP/electron/pull/112) - so you'd just need to use this in your PHP code.

**Comment by danielpetrica** [comment]
Thank you a lot I'll change it this weekend then

**Comment by danielpetrica** [comment]
hi @simonhamp I tried using the newly creatited active method but it returns a null value trigering a type error. while investigating I tried to call  `$this->client->get('screen/active') ` directly inside the Screen class, but it returns an error

```
  {
    "Illuminate\\Http\\Client\\Response": "<!DOCTYPE html>
<html lang=\"en\">
<head>
<meta charset=\"utf-8\">
<title>Error</title>
</head>
<body>
<pre>Cannot GET /api/screen/active</pre>
</body></html>"
  }
```

**Comment by danielpetrica** [comment]
I changed the electron pluign to track dev-main so it's updated and in the vendor I can see the new updates for the active endpoint. I tried running yarn install and also stoping and starting the native:serve command

**Comment by danielpetrica** [comment]
While debbugging I saw that in the server api routes the new routes are missing https://github.com/NativePHP/electron/blob/main/resources/js/electron-plugin/dist/server/api/screen.js
How can I compile it so that it's updated?

**Comment by danielpetrica** [comment]
I've managed to build the electron plugin by running `npm run plugin:build` inside of `vendor/nativephp/electron/resources/js/electron-plugin`.
I can confirm that in my project this code works now, and is much cleaner.

**Comment by simonhamp** [comment]
Glad you found this! I was going to reply earlier with exactly those instructions

Well done 👍🏼

File: src/Facades/Screen.php [hunk 1]
```diff
@@ -7,6 +7,7 @@
 /**
  * @method static object cursorPosition()
  * @method static array displays()
+ * @method static array getCenterOfActiveScreen()
```
Comment (simonhamp) [inline src/Facades/Screen.php:10]: I forgot to add the `active` and `primary` method signatures here - could you do it?

File: src/Screen.php [hunk 2]
```diff
@@ -3,18 +3,54 @@
 namespace Native\Laravel;
 
 use Native\Laravel\Client\Client;
+use Illuminate\Support\Facades\Log;
 
 class Screen
 {
-    public function __construct(protected Client $client) {}
+  public function __construct(protected Client $client)
```
Comment (simonhamp) [inline src/Screen.php:10]: Indentation seems to have changed from 4-space to 2-space throughout this file... can you change it back?

File: src/Screen.php [hunk 3]
```diff
@@ -3,18 +3,54 @@
 namespace Native\Laravel;
 
 use Native\Laravel\Client\Client;
+use Illuminate\Support\Facades\Log;
```
Comment (simonhamp) [inline src/Screen.php:6]: Remove this as we're not using it

File: src/Screen.php [hunk 4]
```diff
@@ -3,18 +3,54 @@
...
+  }
 
-    public function cursorPosition(): object
-    {
-        return (object) $this->client->get('screen/cursor-position')->json();
-    }
+  public function getClient()
+  {
+    return $this->client;
+  }
```
Comment (simonhamp) [inline src/Screen.php:17]: Where is this used?

**Review by simonhamp — CHANGES_REQUESTED** [review]

File: src/Facades/Screen.php (hunk 1 above)
Comment (danielpetrica) [reply src/Facades/Screen.php:10]: yes

File: src/Screen.php (hunk 4 above)
Comment (danielpetrica) [reply src/Screen.php:17]: sorry made that for debug and forgot to clean after myself

**Comment by danielpetrica** [comment]
made the changes required. thanks for pointing the issues out for me

**Review by simonhamp — CHANGES_REQUESTED** [review]
Final thing... please can we return to single-quoted strings where we're not explicitly using interpolation?

Basically this is everywhere you have double-quotes now

**Comment by danielpetrica** [comment]
Yes I'll fix this

**Comment by danielpetrica** [comment]
pushed the update. I'm sorry for all the formatting issues. My IDE applied my default to the project. I'll be more careful next time

**Comment by simonhamp** [comment]
No apology needed. Thanks for putting in the work 🙏🏼

**Review by simonhamp — APPROVED** [review]

**Pull request merged**